arangod/VocBase/Methods/Queries.cpp

```diff
@@ -0,0 +1,188 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2017 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Jan Steemann
+////////////////////////////////////////////////////////////////////////////////
+
+#include "Queries.h"
+
+#include "ApplicationFeatures/ApplicationServer.h"
+#include "Aql/Query.h"
+#include "Aql/QueryExecutionState.h"
+#include "Aql/QueryList.h"
+#include "Auth/TokenCache.h"
+#include "Basics/Common.h"
+#include "Basics/Exceptions.h"
+#include "Cluster/ClusterFeature.h"
+#include "Cluster/ClusterInfo.h"
+#include "Cluster/ServerState.h"
+#include "GeneralServer/AuthenticationFeature.h"
+#include "Network/NetworkFeature.h"
+#include "Network/Methods.h"
+#include "Network/Utils.h"
+#include "VocBase/vocbase.h"
+
+#include <velocypack/Builder.h>
+#include <velocypack/Iterator.h>
+#include <velocypack/Slice.h>
+#include <velocypack/velocypack-aliases.h>
+
+using namespace arangodb;
+using namespace arangodb::methods;
+
+namespace {
+
+network::Headers buildHeaders() {
+  auto auth = AuthenticationFeature::instance();
+
+  network::Headers headers;
+  if (auth != nullptr && auth->isActive()) {
+    headers.try_emplace(StaticStrings::Authorization,
+                        "bearer " + auth->tokenCache().jwtToken());
+  }
+  return headers;
+}
+
+/// @brief return the list of currently running queries
+void getQueries(TRI_vocbase_t& vocbase, std::vector<aql::QueryEntryCopy> const& queries,
+                velocypack::Builder& out, char const* action, bool fanout) {
+  out.openArray();
+  
+  // local case
+  for (auto const& q : queries) {
+    q.toVelocyPack(out);
+  }
+
+  if (ServerState::instance()->isCoordinator() && fanout) {
```

Comment:
Should this just be `if (fanout)` is there a case where we would have set `fanout == true` for a single server?

Reply:
I agree in general, but it is possible to trigger the fanout variable value via a URL parameter sent to the REST handler. So I would like to additionally restrict this to coordinators in case the URL parameter is tainted.

Reply:
That's a fair point.